src/test/java/io/strimzi/kafka/metrics/YammerMetricsCollectorTest.java

```diff
@@ -48,35 +48,73 @@ public void testCollect() {
-        assertEquals(0.0, metrics.get(0).samples.get(0).value, 0.1);
-        assertEquals(new ArrayList<>(tags.keySet()), metrics.get(0).samples.get(0).labelNames);
-        assertEquals(new ArrayList<>(tags.values()), metrics.get(0).samples.get(0).labelValues);
+        assertEquals(1, metricFamilySamples.samples.size());
+
+        Collector.MetricFamilySamples.Sample serverGroupNameSamples = metricFamilySamples.samples.get(0);
+        assertEquals(0.0, serverGroupNameSamples.value, 0.1);
+        assertEquals(new ArrayList<>(tags.keySet()), serverGroupNameSamples.labelNames);
+        assertEquals(new ArrayList<>(tags.values()), serverGroupNameSamples.labelValues);
 
         // Updating the value of the metric
         counter.inc(10);
         metrics = collector.collect();
         assertEquals(1, metrics.size());
-        assertEquals("kafka_server_group_name_type_count", metrics.get(0).name);
-        assertEquals(1, metrics.get(0).samples.size());
-        assertEquals(10.0, metrics.get(0).samples.get(0).value, 0.1);
+
+        metricFamilySamples = metrics.get(0);
+        serverGroupNameSamples = metricFamilySamples.samples.get(0);
+
+        assertEquals("kafka_server_group_name_type_count", metricFamilySamples.name);
+        assertEquals(1, metricFamilySamples.samples.size());
+        assertEquals(10.0, serverGroupNameSamples.value, 0.1);
 
         // Removing the metric
         removeMetric("group", "name", "type");
         metrics = collector.collect();
         assertTrue(metrics.isEmpty());
     }
 
+    @Test
+    public void testCollectNonNumericMetric() {
+        Map<String, String> props = new HashMap<>();
+        props.put(PrometheusMetricsReporterConfig.ALLOWLIST_CONFIG, "kafka_server_group_name.*");
+        PrometheusMetricsReporterConfig config = new PrometheusMetricsReporterConfig(props);
+        YammerMetricsCollector collector = new YammerMetricsCollector(config);
+
+        List<Collector.MetricFamilySamples> metrics = collector.collect();
+        assertTrue(metrics.isEmpty());
+
+        newNonNumericGauge("group", "name", "type");
+        metrics = collector.collect();
+
+        Map<String, String> expectedTags = new LinkedHashMap<>(tags);
+        expectedTags.put("type", "value");
+        assertEquals(1, metrics.size());
+
+        Collector.MetricFamilySamples metricFamilySamples = metrics.get(0);
+
+        assertEquals("kafka_server_group_name_type", metricFamilySamples.name);
+        assertEquals(1, metricFamilySamples.samples.size());
+        assertMetricFamilySample(metricFamilySamples, "kafka_server_group_name_type", 1.0, expectedTags);
+    }
+
+    private void assertMetricFamilySample(Collector.MetricFamilySamples actual, String expectedSampleName, double expectedValue, Map<String, String> expectedTags) {
+        assertEquals(expectedSampleName, actual.name, "unexpected name");
+        assertEquals(1, actual.samples.size(), "unexpected number of samples");
```

Comment:
It looks like we already checked have these 2 assertions on line 102-103. Why are we checking these twice?

Reply:
Also this is only called from a single place, so maybe this can be inlined?

Reply:
Duplicates removed and changed to inline